pontoon/sync/changeset.py

```diff
@@ -104,6 +105,12 @@ def execute_update_vcs(self):
             # Track which translators were involved.
             self.commit_authors_per_locale[locale_code].extend([t.user for t in db_translations if t.user])
 
+        # Remove obsolete entities from VCS
+        for r in Resource.objects.filter(entities__pk__in=self.changes['obsolete_db']).distinct():
```

Comment:
I would use here values_list() or defer() to save some memory. Also, we could create an method in queryset that could easily return assymetric resources, instead of calling property each time.